### DISCUSSION TITLE
Can't enable additional plugins when container started with bind mounted enabled_plugins file

### DISCUSSION THREAD

**Discussion by al3xsun** [docker-library/rabbitmq#497] [Q&A] [answered]
Hi,
I am using 3.8.11 image, if I start container by command like below
`docker container run ... --mount type=bind,src=/enabled_plugins,dst=/etc/rabbitmq/enabled_plugins ...`
My enabled_plugins file contains the initial list of plugins I want to enable, but after the container starts, I try to run 
`rabbitmq-plugins enable <tier-1 plugin>` for some tier-1 plugin that's not enabled, I get 

`rabbitmq-plugins enable rabbitmq_prometheus
rabbitmq_prometheus
Error:
{:cannot_write_enabled_plugins_file, '/etc/rabbitmq/enabled_plugins', :ebusy}`

the file permission for the external file is and I believe that's shared with docker container:
`-rw-r--r-- 1 root root`
and even if I change the file permission to 
`-rw-rw-rw- 1 root root`
I still get the same error when trying to enable an additional plugin. 

I am wondering whether this is expected from docker mounted enabled_plugin files? As if I try running with --offline command it still can't change the enabled_plugin file, so the only way to enable more plugins is to change the file enable the plugins upon starting the rabbitmq container?

Really appreciate if anyone can point out whether it's the bind mount that cause this issue or I am doing something wrong, apologize in advance if I missed something obvious!

**Comment by tianon** [comment]
This is very likely the same issue that `sed -i` has with bind-mounted files -- when you bind-mount a file, it bind-mounts that specific inode within the filesystem, so the standard method of "write new file, then rename to overwrite the old file with the new file atomically" doesn't work unless you mount a full directory instead.

**Comment by al3xsun** [comment]
Thanks for your response! 

Then to resolve this issue, should I mount some thing like:
`--mount type=bind,src=/etc/<folder that contains enabled_plugins file>,dst=/etc/rabbitmq/` 
Since I mount other files (conf) to the /etc/rabbitmq, and upon checking the /etc/rabbitmq directory inside my container, it contains other files that i didn't mount specifically, so this might not be ideal? I might be messing up what's in this rabbitmq directory.

I am not sure whether change the mount type would help? (Change to volume, tmpfs mount?) Appreciate if you have any suggestions!

**Comment by yosifkit** [comment]
>I am not sure whether change the mount type would help?

Mounting files or directories from the host need to be be `type=bind`; it's the only way to do it.

**Comment by al3xsun** [comment, marked as answer]
Seems like the way out is to mount a directory that contains the enabled plugin files I would like to change, + override the RABBITMQ_ENABLED_PLUGINS_FILE, will try that out, thx a lot for the quick response!
